native_client/deepspeech.cc

```diff
@@ -448,6 +540,62 @@ DS_CreateModel(const char* aModelPath,
 
   *retval = model.release();
   return tensorflow::error::OK;
+#else // USE_TFLITE
+  TfLiteStatus status;
+
+  model->fbmodel = tflite::FlatBufferModel::BuildFromFile(aModelPath);
+  if (status != kTfLiteOk) {
+    std::cerr << status << std::endl;
+    return status;
+  }
+
+
+  tflite::ops::builtin::BuiltinOpResolver resolver;
+  status = tflite::InterpreterBuilder(*model->fbmodel, resolver)(&model->interpreter);
+  if (status != kTfLiteOk) {
+    std::cerr << status << std::endl;
+    return status;
+  }
+
+  model->interpreter->AllocateTensors();
+  model->interpreter->SetNumThreads(4);
+
+  TfLiteIntArray* dims_input_node = model->interpreter->tensor(model->interpreter->inputs()[0])->dims;
+
+  model->n_steps = dims_input_node->data[1];
+  model->n_context = (dims_input_node->data[2] - 1 ) / 2;
+  model->mfcc_feats_per_timestep = dims_input_node->data[2] * dims_input_node->data[3];
+
+  TfLiteIntArray* dims_logits = model->interpreter->tensor(model->interpreter->outputs()[0])->dims;
+  const int final_dim_size = dims_logits->data[1] - 1;
+  if (final_dim_size != model->alphabet->GetSize()) {
+    std::cerr << "Error: Alphabet size does not match loaded model: alphabet "
+              << "has size " << model->alphabet->GetSize()
+              << ", but model has " << final_dim_size
+              << " classes in its output. Make sure you're passing an alphabet "
+              << "file with the same size as the one used for training."
+              << std::endl;
+    return EINVAL;
+  }
+
+  const int previous_state_c_id = model->interpreter->inputs()[1];
+  const int previous_state_h_id = model->interpreter->inputs()[2];
+
+  TfLiteIntArray* dims_c = model->interpreter->tensor(previous_state_c_id)->dims;
+  TfLiteIntArray* dims_h = model->interpreter->tensor(previous_state_h_id)->dims;
+  assert(dims_c->data[1] == dims_h->data[1]);
+
+  model->previous_state_size = dims_c->data[1];
+  model->previous_state_c_ = new float[model->previous_state_size]();
```

Comment:
Use an `std::vector<float>` or at least an `std::unique_ptr<float[]>`.

Reply:
Done, testing the changes